src/system/flow.rs

```diff
@@ -27,74 +27,110 @@ use async_std::{
     task,
 };
 use hashbrown::HashMap;
-use std::{borrow::Borrow, collections::HashSet};
+use std::collections::HashSet;
 use std::{sync::atomic::Ordering, time::Duration};
 use tremor_common::ids::{ConnectorIdGen, OperatorIdGen};
 use tremor_script::{
-    ast::{self, ConnectStmt, DeployEndpoint, DeployFlow, Helper},
+    ast::{self, ConnectStmt, DeployFlow, Helper},
     errors::not_defined_err,
 };
 
 /// unique identifier of a flow instance within a tremor instance
-#[derive(Debug, PartialEq, PartialOrd, Eq, Hash, Clone, Serialize)]
-pub(crate) struct Id(pub(crate) String);
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, PartialEq, PartialOrd, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct FlowAlias(String);
```

Comment:
Should the flow alias perhaps include it's "type" (or definition name) so that it's easier to distinguish where it came from? It could render to something like `"my_alias<my_defition>"` or `"my_definition::my_alias"` or a bunch of other things, but having this type would allow to convey both, the definition name and the created alias for it

Reply:
This is a good idea. this will make tracking instances from metrics back to their definition much easier. We already use the `::` for separating the hierarchical parts of flow and pipeline/connector `flow_alias::pipeline_alias`, so this should be printed differently, maybe: `flow_alias::pipeline_alias from pipeline_definition_name` ? Or `pipeline_definition_name<flow_alias::pipeline_alias>`?

Reply:
Ja, you've got a point `::` is a bad choice, while `<>` reads kind of "right" from the rust perspective I honestly like your suggestion of `from` as it mirrors the way we define things :+1: for consistency :)

Reply:
I tried it and it gets really too long for logs to remain readable. I would add this as another issue and discuss it there?